pymc3_hmm/utils.py

```diff
@@ -458,3 +457,59 @@ def plot_predictive_histograms(
             )
 
     return axes
+
+
+def plot_split_timeseries_hists(
+    plot_data: pd.DataFrame,
+    sample_col: str,
+    sample_mean_col: str,
+    twin_column_name: Union[str, None] = None,
+    observed_col: Union[str, None] = None,
+    **split_ts_kwargs
+):  # pragma: no cover
+    from matplotlib.ticker import MaxNLocator
+
+    def plot_fn(ax, data, **kwargs):
+        if observed_col is not None:
+            ax.plot(
+                data[observed_col],
+                label=observed_col,
+                alpha=0.7,
+                color="red",
+                linewidth=0.8,
+                drawstyle="steps",
+            )
+            # make sample mean less visible
+            alpha, color = 0.3, "blue"
+        else:
+            # make sample mean more visible
+            alpha, color = 0.7, "darkorange"
+
+        ax.plot(
+            data[sample_mean_col],
+            label=sample_mean_col,
+            alpha=alpha,
+            color=color,
+            linewidth=0.8,
+            drawstyle="steps",
+        )
+
+        plot_predictive_histograms(data[sample_col], axes=ax)
+
+    axes_split_data = plot_split_timeseries(
+        plot_data,
+        twin_column_name=twin_column_name,
+        plot_fn=plot_fn,
+        **split_ts_kwargs,
+    )
+
+    if twin_column_name is not None:
+        for (_, twin_ax), _ in axes_split_data:
+            _ = twin_ax.set_ylabel(
+                twin_column_name, color=twin_ax.get_lines()[0].get_color()
+            )
+            _ = twin_ax.yaxis.set_major_locator(MaxNLocator(integer=True))
+
+    plt.tight_layout()
```

Comment:
Similar to what I said earlier, this is a very specific thing that can be done by the caller.

Reply:
Ah, the reason why I left this at the bottom is that it uses the `twin_ax` variable which comes out of the `plot_split_timeseries ` function. Do we not want to use the `twin_column_name` parameter in that function and does something like that in `plot_fn`?

Reply:
`plt.tight_layout` might be a strong default choice.  It make some changes to a plot that a user might not want.

Also, the twin column line plot with `MaxNLocator` is too specific of a choice to be included in a generic plot-split-timeseries-histogram function.